charmcraft/package.py

```diff
@@ -89,7 +91,10 @@ def __init__(self, *, config, force, debug, shell, shell_after, measure):
         self.charmdir = config.project.dirpath
         self.buildpath = self.charmdir / BUILD_DIRNAME
         self.config = config
-        self._parts = self.config.parts.copy()
+        if self.config.parts:
+            self._parts = self.config.parts.copy()
+        else:
+            self._parts = None
```

Comment:
Is `config.parts` undefined or defined as an empty dictionary if no parts are defined by the user? In the former case, how was this working before?